src/local/ample.lean

```diff
@@ -13,9 +15,90 @@ vector spaces with their natural topology.
 
 open set
 
-variables {F : Type*} [add_comm_group F] [vector_space ℝ F] [topological_space F]
+variables {E : Type*} [add_comm_group E] [vector_space ℝ E] [topological_space E]
 
-/-- A subset of a topological real vector space is ample if the convex hull of each of its 
+/-- A subset of a topological real vector space is ample if the convex hull of each of its
 connected components is the full space. -/
-def ample_set (s : set F) := 
-∀ x : s, convex_hull (subtype.val '' (connected_component x)) = univ
\ No newline at end of file
+def ample_set (s : set E) :=
+∀ x : s, convex_hull (subtype.val '' (connected_component x)) = univ
+
+local notation ` I ` := interval (0 : ℝ) 1
+
+def path_connected {E : Type*} [topological_space E] {F : set E} (x y : F) : Prop :=
+∃ (f : I → F), continuous f ∧ f ⟨0, left_mem_interval⟩ = x ∧ f ⟨1, right_mem_interval⟩ = y
+
+def path_connected_set {E : Type*} [topological_space E] (F : set E) : Prop :=
+∀ (x y : F), path_connected x y
+
+def flip_interval : I → I := λ x,
+{ val := 1 - x.1,
+  property :=
+  begin
+    rcases x with ⟨_, _, _⟩,
+    norm_num at *,
+    split; linarith
+  end }
+
+lemma continuous_flip : continuous flip_interval := sorry
+
+lemma path_connected_equivalence {E : Type*} [topological_space E] (F : set E) :
```

Comment:
I think it's better to separately prove the `@[refl]`, `@[symm]` and `@[trans]` lemmas, because they will also be useful for tactics like `symmetry` and `transitivity`.

Reply:
Good point, thanks!